apps/wallet-dashboard/tests/connect.spec.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { test, expect } from './fixtures';
+import { createWallet } from './utils';
+import 'dotenv/config';
+
+test.describe('Wallet Connection', () => {
+    test.beforeEach(async ({ page, extensionUrl, sharedState }) => {
+        // Navigate to the wallet dashboard
+        await page.goto('/');
+        await page.waitForSelector('.welcome-page');
+
+        // Import a wallet in the extension
+        await page.goto(extensionUrl);
+        const cratedWallet = await createWallet(page, extensionUrl);
+
+        sharedState.walletMnemonic = cratedWallet.mnemonic || '';
+        sharedState.walletAddress = cratedWallet.address || '';
+
+        // Go back to dashboard
+        await page.goto('/');
+        await page.waitForSelector('.welcome-page');
+    });
+
+    test('should connect to wallet extension', async ({ context, page, sharedState }) => {
+        const connectButton = page.getByRole('button', { name: 'Connect' });
+        await connectButton.click();
+
+        // Select the extension wallet option
+        await page.getByText('IOTA Wallet', { exact: true }).click();
```

Comment:
not sure about this exact because local can use the DEV one too, right?

Reply:
but when testing locally we are most likely always using the dev version, so i suggest setting the extension name in the fixtures so that you can use it in the tests :)

Reply:
But that will make the tests slower locally, won't it? In my opinion it shouldn't be required to build the wallet but I guess we can go with your approach